src/bit.rs

```diff
@@ -48,12 +50,20 @@ pub fn bits_to_byte(bits: [bool; 8]) -> u8 {
 }
 
 /// Converts a vector of input bits (little-endian) to its integer representation
-pub fn bits_to_uint(bits: &[bool]) -> usize {
-    let mut value: usize = 0;
-    for i in 0..bits.len() {
-        value |= (bits[i] as usize) << i;
+/// Returns None if the length of `bits` is greater than `usize`, which would cause an attempt to shift left with
+/// overflow
+pub fn bits_to_uint(bits: &[bool]) -> Option<usize> {
```

Comment:
Think we should keep the naming for these functions consistent. I chose to name the inverse of this function [checked_uint_to_bits](https://github.com/tari-project/tari_utilities/pull/9) because of the `checked_sub`, `checked_add`, `checked_xxxx` etc naming in rust stdlib when returning an Option on overflows. I don't have a strong preference and happy to change my PR. Thoughts?

Reply:
Yeah I agree with your name change I will update